### ISSUE TITLE
S1144 FP: tuple construction should be considered for member usage

### ISSUE THREAD

**Issue by nnpcYvIVl** [SonarSource/sonar-dotnet#2333]
The following code flags [S1144](https://rules.sonarsource.com/csharp/RSPEC-1144) on the Name setter but when I remove it compilation fails with CS0200.

```
namespace S1144_False_Positive_3
{
    class Class
    {
        void M()
        {
            OtherClass oc = new OtherClass();
            (oc.Name, oc.Type) = ("A", "B");
        }

        class OtherClass
        {
            public string Name { get; set; }
            public string Type { get; set; }
        }
    }
}
```

Is this a false positive?

### Related information

* SonarC# Version : Latest
* Visual Studio Version: 15.9.8

**Comment by Corniel** [comment]
To me this is indeed a false positive. The `TupleExpression` nodes are not taken into account.

``` C#
namespace SonarAnalyzer.Experiments.CSharp
{
    public class S1144
    {
        public void M()
        {
            OtherClass oc = new OtherClass();
            (oc.Name, oc.Type) = ("A", "B");
        }

        class OtherClass
        {
            public string Name { get; set; } // Compliant, usage via a TupleExpression should be recognized.
            public string Type { get; set; } // Compliant, usage via a TupleExpression should be recognized.
        }
    }
}
```

**Comment by andrei-epure-sonarsource** [comment]
thanks @nnpcYvIVl  and @Corniel  - I confirm the FP
